client/state/themes/actions.js

```diff
@@ -404,6 +407,43 @@ export function themeActivated( themeStylesheet, siteId, source = 'unknown', pur
 }
 
 /**
+ * Triggers a network request to install WordPress.com theme on Jetpack site.
+ * Requires Jetpack 4.4
+ *
+ * @param  {String}   siteId       Jetpack Site ID
+ * @param  {String}   wpcomThemeId WP.com Theme ID with -wpcom suffix
+ * @return {Function}              Action thunk
+ */
+export function installWpcomThemeOnJetpack( siteId, wpcomThemeId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: THEME_INSTALL_ON_JETPACK_REQUEST,
+			siteId,
+			wpcomThemeId
+		} );
+
+		return wpcom.undocumented().installThemeOnJetpack( siteId, wpcomThemeId )
+			.then( ( ) => {
+				dispatch( {
```

Comment:
I am adding it but It looks like we do only receive limited subset of theme data from the endpoint.